root/etc/s6-overlay/s6-rc.d/svc-cron/run

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/with-contenv bash
+
+# Install DockerCLI if true
+if [ "${DOCKERCLI:-}" == "true" ]; then
+  echo "Installing Docker CLI..."
+  apk add -U --quiet docker-cli
+  dockerver=$(docker --version | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+')
+else
+  dockerver="not installed"
+fi
+
+# Version variables
+borgver=$(borg --version)
+borgmaticver=$(borgmatic --version)
+apprisever=$(apprise --version | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+')
+pythonver=$(python3 --version | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+')
+
+# Software versions
+echo "Software Versions:
+-----------------------------------
+apprise $apprisever
+$borgver
+borgmatic $borgmaticver
+dockercli $dockerver
+python $pythonver
+-----------------------------------
+Time Zone: $TZ
+-----------------------------------
+"
+
+# Enable initial debug logging based on the DEBUG_SECRETS environment variable.
+# Logs the initial values of BORG_PASSPHRASE and BORG_PASSPHRASE_FILE.
+if [ "${DEBUG_SECRETS}" = "true" ] || [ "${DEBUG_SECRETS}" = "1" ]; then
+  echo "Before: BORG_PASSPHRASE: ${BORG_PASSPHRASE}"
+  echo "Before: BORG_PASSPHRASE_FILE: ${BORG_PASSPHRASE_FILE}"
+  echo "Before: YOUR_PASSPHRASE: ${YOUR_PASSPHRASE}"
+  echo "Before: YOUR_PASSPHRASE_FILE: ${YOUR_PASSPHRASE_FILE}"
+fi
+
+# Loop through all environment variables that start with 'BORG'.
+for var_name in $(set | grep -E '^BORG|^YOUR' | awk -F= '{print $1}'); do
+  # Retrieve the current value of each environment variable.
+  var_value=$(eval echo \$$var_name)
+
+  # Check if the variable's name ends with '_FILE'.
+  if [[ "$var_name" =~ _FILE$ ]]; then
+    # Strip the '_FILE' suffix to obtain the corresponding variable name.
+    original_var_name=${var_name%_FILE}
+
+    # Retrieve the value of the original environment variable, if it exists.
+    original_var_value=$(eval echo \$$original_var_name)
+
+    # Ensure the *_FILE variable is valid, and the referenced file exists and is not empty.
+    if [ -n "$var_value" ] && [ -s "$var_value" ]; then
+      # Notify user if original variable is being overwritten.
+      if [ -n "$original_var_value" ]; then
+        echo "Note: $original_var_name was already set but is being overwritten by $var_name"
+      fi
+
+      # Update the original variable with the content of the file.
+      export "$original_var_name"=$(cat "$var_value")
```

Comment:
I might be missing something, but how do these environment variable exports make their way into the borgmatic invocation that cron runs?

Reply:
@Psycho0verload - one for you

Reply:
@witten is right.

Unfortunately, I am not familiar with S6 either. That's why I found it easiest to integrate the script into the script call, which runs in any case.

I have looked at S6 and probably the best solution would be to make a separate call for the secrets. @modem7 maybe you are more familiar with this and can build your own call for it. The script itself works. 

Otherwise I would experiment with it.

Reply:
For now, I would say let's keep it simple. If the script works for now, then that's all we need. Once we're happy with how the full PR is, and that it works as intended with no bugs from the migration, we can start investigating S6 in its entirety, and look at expanding it with additional calls.

We'll be a lot less confined with S6, and we can add/remove things in a modular fashion going forwards, making it much easier to test and develop.

Reply:
Are we all happy with this then?

Reply:
LGTM

Reply:
Will await @Psycho0verload 's review then I'll tick and we can merge this bad boy